hnswlib/hnswalg.h

```diff
@@ -665,7 +690,7 @@ class HierarchicalNSW : public AlgorithmInterface<dist_t> {
         auto pos = input.tellg();
 
         /// Optional - check if index is ok:
-        input.seekg(cur_element_count * size_data_per_element_, input.cur);
+        input.seekg(cur_element_count * size_data_per_element_ + cur_element_count * sizeof(float), input.cur);
```

Comment:
This seeks to the linked lists and makes sure they are laid out correctly. Since we add data, we have to seek to it.